GitHub.Collectors/Web/GitHubRateLimiter.cs

```diff
@@ -54,13 +52,22 @@ protected override async Task WaitIfNeededAsync(IAuthentication authentication,
                     };
                     this.TelemetryClient.TrackEvent("RateLimiterDelay", properties);
 
+                    if (this.throwOnRateLimit)
+                    {
+                        Exception e = new Exception("RateLimitRequeue");
```

Comment:
I would create a dedicated class (e.g., GitHubRateLimitException) rather than utilizing the "data" column of exception here.